### PR TITLE
✨ update exposures when changing margin account.

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

File: src/accounts/AccountV1.sol [hunk 1]
```diff
@@ -266,23 +266,43 @@ contract AccountV1 is AccountStorageV1, IAccount {
...
+     * @param newCreditor The contract address of the Creditor.
      * @dev Currently only one Creditor can be set
      * (we are working towards a single account for multiple Creditors tho!).
      * @dev Only open margin accounts for protocols you trust!
      * The Creditor has significant authorization: use margin, trigger liquidation, and manage assets.
      */
-    function openMarginAccount(address creditor_) external onlyOwner {
-        if (creditor != address(0)) revert AccountErrors.CreditorAlreadySet();
-        _openMarginAccount(creditor_);
+    function openMarginAccount(address newCreditor) external onlyOwner notDuringAuction {
```
Comment (j-vp) [inline src/accounts/AccountV1.sol:275]: think this also needs a nonreentrant. If we're calling a random address on ICreditor(new/oldCreditor), they may call back this same function (for example if the given creditor is the account owner), and thus go through the batchProcessDep/With multiple times?

Comment (Thomas-Smets) [reply src/accounts/AccountV1.sol:275]: we should list every function that calls unchecked addresses

**Review by j-vp — APPROVED** [review]
gtg

File: src/accounts/AccountV1.sol [hunk 2]
```diff
@@ -266,23 +266,45 @@ contract AccountV1 is AccountStorageV1, IAccount {
...
+        // Check if all assets in the Account are allowed by the new Creditor
+        // and add the exposure of the account for the new Creditor.
+        IRegistry(registry).batchProcessDeposit(newCreditor, assetAddresses, assetIds, assetAmounts);
+
+        // Remove the exposures of the Account for the old Creditor.
+        if (oldCreditor != address(0)) {
+            IRegistry(registry).batchProcessWithdrawal(oldCreditor, assetAddresses, assetIds, assetAmounts);
+        }
+
+        // Close the active margin account for the old Creditor.
```
Comment (Thomas-Smets) [inline src/accounts/AccountV1.sol:292]:
```suggestion
        // Open margin account with the new Creditor.
```

**Pull request merged**